src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,88 @@
-export const Pagination = () => {};
+import React from 'react';
+import classNames from 'classnames';
+import { getNumbers } from '../../utils';
+
+type Props = {
+  total: number,
+  perPage: number,
+  currentPage: number,
+  onPageChange: (page: number | string) => void;
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const lastPage = Math.ceil(total / perPage);
+  const amountOfPage = getNumbers(1, lastPage);
+
+  return (
+    <>
+      <ul className="pagination">
+        <li
+          className={classNames(
+            'page-item',
+            { disabled: currentPage === 1 },
+          )}
+        >
+          <a
+            data-cy="prevLink"
+            className="page-link"
+            href="#prev"
+            aria-disabled={currentPage === 1}
+            onClick={() => {
+              if (currentPage !== 1) {
+                onPageChange('prev');
+              }
+            }}
+          >
+            «
+          </a>
+        </li>
+        {amountOfPage.map(page => (
+          <li
+            key={page}
+            className={classNames(
+              'page-item',
+              { active: page === currentPage },
+            )}
+          >
+            <a
+              data-cy="pageLink"
+              className="page-link"
+              href={`#${page}`}
+              onClick={() => {
+                onPageChange(page);
+              }}
+            >
+              {page}
+            </a>
+          </li>
+        ))}
+
+        <li
+          className={classNames(
+            'page-item',
+            { disabled: currentPage === lastPage },
+          )}
+        >
+          <a
+            data-cy="nextLink"
+            className="page-link"
+            href="#next"
+            aria-disabled={currentPage === lastPage}
+            onClick={() => {
+              if (currentPage !== lastPage) {
+                onPageChange('next');
```

Comment:
Instead of calling `onPageChange('next')`, consider calling `onPageChange(currentPage + 1)` for better clarity and consistency with the function's argument type.